internal/templating/services/templates_pvc.go

```diff
@@ -51,106 +52,215 @@ func GeneratePVCTemplate(
+) (*corev1.PersistentVolumeClaim, error) {
+	if serviceValues.PersistentVolumeName != "" {
+		if serviceValues.PersistentVolumeName != serviceValues.OverrideName {
+			// this service base volume is not needed because it is created by a different service
+			// lagoon legacy templates allowed due to a funny templating issue, for multiple "basic" types to mount one volume
+			// from one main service, across multiple services of the same type
+			return nil, nil
+		}
+	}
+	serviceTypeValues := &servicetypes.ServiceType{}
+	helpers.DeepCopy(val, serviceTypeValues)
+	persistentVolumeSize := serviceTypeValues.Volumes.PersistentVolumeSize
+	if serviceValues.PersistentVolumeSize != "" {
+		persistentVolumeSize = serviceValues.PersistentVolumeSize
+	}
+	serviceType := &servicetypes.ServiceType{}
+	helpers.DeepCopy(val, serviceType)
+
+	additionalLabels := map[string]string{}
+	additionalAnnotations := map[string]string{}
+
+	additionalLabels["app.kubernetes.io/name"] = serviceType.Name
+	additionalLabels["app.kubernetes.io/instance"] = serviceValues.OverrideName
+	additionalLabels["lagoon.sh/template"] = fmt.Sprintf("%s-%s", serviceType.Name, "0.1.0")
+	additionalLabels["lagoon.sh/service"] = serviceValues.OverrideName
+	additionalLabels["lagoon.sh/service-type"] = serviceType.Name
+
+	additionalAnnotations["k8up.syn.tools/backup"] = strconv.FormatBool(serviceTypeValues.Volumes.Backup)
+	additionalAnnotations["k8up.io/backup"] = strconv.FormatBool(serviceTypeValues.Volumes.Backup)
+
+	pvc := &corev1.PersistentVolumeClaim{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "PersistentVolumeClaim",
+			APIVersion: corev1.SchemeGroupVersion.Version,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: serviceValues.OverrideName,
+		},
+	}
+
+	if serviceTypeValues.Volumes.PersistentVolumeType == corev1.ReadWriteMany {
+		pvc.Spec.StorageClassName = helpers.StrPtr("bulk")
+	}
+
+	// add any remaining changes that are shared between default and additional
+	err := updatePVC(
+		pvc,
+		&buildValues,
+		serviceValues.OverrideName,
+		persistentVolumeSize,
+		serviceTypeValues.Volumes.PersistentVolumeType,
+		labels, annotations,
+		additionalLabels, additionalAnnotations,
+	)
+	if err != nil {
+		return nil, err
+	}
+	// end PVC template
+	return pvc, nil
+}
+
+// generateAdditionalPVC additional volumes have different labels/annotations to a default, and also handle some configuration a bit differently
+func generateAdditionalPVC(
+	buildValues generator.BuildValues,
+	additionalVolume generator.ComposeVolume,
+	labels, annotations map[string]string,
+) (*corev1.PersistentVolumeClaim, error) {
+	additionalLabels := map[string]string{}
+	additionalAnnotations := map[string]string{}
+
+	additionalLabels["app.kubernetes.io/name"] = lagoon.GetVolumeNameFromLagoonVolume(additionalVolume.Name)
+	additionalLabels["app.kubernetes.io/instance"] = additionalVolume.Name
+	additionalLabels["lagoon.sh/template"] = fmt.Sprintf("%s-%s", "additional-volume", "0.1.0")
```

Comment:
yeah, if this is the same version as the other `0.1.0` then popping in a const def makes sense.